config-object-store/src/main/java/org/hypertrace/config/objectstore/IdentifiedFilterPushedDownObjectStore.java

```diff
@@ -58,6 +72,40 @@ public Optional<T> getMatchingData(RequestContext context, F filterInput, List<S
     return getMatchingObject(context, filterInput, sortInput).map(ConfigObject::getData);
   }
 
+  List<ContextualConfigObject<T>> getMatchingObjectsWithFilter(
+      RequestContext context, Filter filter, List<SortBy> sortByList, Pagination pagination) {
+    return context
+        .call(
+            () ->
+                this.configServiceBlockingStub
+                    .withDeadline(getDeadline())
+                    .getAllConfigs(buildGetAllConfigsRequest(filter, sortByList, pagination)))
+        .getContextSpecificConfigsList()
+        .stream()
+        .map(
+            contextSpecificConfig ->
+                ContextualConfigObjectImpl.tryBuild(
+                    contextSpecificConfig, this::buildDataFromValue))
+        .flatMap(Optional::stream)
+        .collect(
+            Collectors.collectingAndThen(
+                Collectors.toUnmodifiableList(), this::orderFetchedObjects));
+  }
+
+  private GetAllConfigsRequest buildGetAllConfigsRequest(
+      Filter filter, List<SortBy> sortByList, Pagination pagination) {
```

Comment:
nit - annotate nullability (especially where these same values show up on the public part of the API) since we're inconsistent here, pagination can be null, filter + sort by list can't be.

Reply:
done

Reply:
I wish we did this in every method, but at a minimum every method in this callstack that has inconsistent treatment for different arguments.

Reply:
Updated in all the methods where we use pagination.